libvast/vast/plugin.hpp

```diff
@@ -304,6 +317,14 @@ class plugin_ptr final {
 
 #if defined(VAST_ENABLE_STATIC_PLUGINS_INTERNAL)
 
+#  if VAST_ENABLE_STATIC_PLUGINS_INTERNAL == 1
```

Comment:
It feels like there should be a cleaner solution to this than having a magic number.

How about adding an additional defines like `VAST_OPTIONAL_PLUGIN`/`VAST_NATIVE_PLUGIN` that are set in the respective build contexts?

Or alternatively we at least could turn the numbers into named constants like
```
#define PLUGIN_VISIBILITY_DYNAMIC 1
```
etc.

Reply:
The logic is a bit more readable now and does not rely on magic constants any more.